### PR TITLE
Add managed identity auth

### PR DESCRIPTION
**Pull request by shoudusse**
This pull request allows user to choice [Azure Managed Identities](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview) for authentication

### CODE REVIEW

**Comment by shoudusse** [comment]
This PR fixes #61

File: README.md [hunk 1]
```diff
@@ -56,6 +60,8 @@ If you want to scrape metrics from Azure national clouds (e.g. AzureChinaCloud,
 If you won't provide `active_directory_authority_url` and `resource_manager_url` parameters, azure-metrics-exporter scrapes metrics from global cloud.
 You can find endpoints for national clouds [here](http://www.azurespeed.com/Information/AzureEnvironments)
 
+If using managed identities, only `subscription_id` is needed.
```
Comment (brian-brazil) [inline README.md]: This is a bit throwaway, I can see it causing confusion. I'd suggest stating up top which way this is.

File: main.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ var (
 	configFile            = kingpin.Flag("config.file", "Azure exporter configuration file.").Default("azure.yml").String()
 	listenAddress         = kingpin.Flag("web.listen-address", "The address to listen on for HTTP requests.").Default(":9276").String()
 	listMetricDefinitions = kingpin.Flag("list.definitions", "List available metric definitions for the given resources and exit.").Bool()
+	managedIdentity       = kingpin.Flag("managed.identity", "Use managed identity to authenticate to Azure API.").Bool()
```
Comment (brian-brazil) [inline main.go]: This should come from the config, not the command line.

File: azure.go [hunk 3]
```diff
@@ -183,14 +183,27 @@ func NewAzureClient() *AzureClient {
...
-	}
-	resp, err := ac.client.PostForm(target, form)
+	var resp *http.Response
+	var err error
+	if *managedIdentity {
+		log.Printf("Using managed identity")
+		target := fmt.Sprintf("http://169.254.169.254/metadata/identity/oauth2/token?resource=%s&api-version=2018-02-01", sc.C.ResourceManagerURL)
+		req, err := http.NewRequest("GET", target, nil)
+		if err != nil {
+			return fmt.Errorf("Error authenticating against Azure API: %v", err)
```
Comment (brian-brazil) [inline azure.go]: Is there actually any auth going on here?

File: README.md [hunk 4]
```diff
@@ -37,9 +37,13 @@ This exporter requires a configuration file. By default, it will look for the az
 
 This exporter reads metrics from an existing Azure subscription with these requirements:
 
+* If not using managed identities:
   * An application must be registered (e.g., Azure Active Directory -> App registrations -> New application registration)
   * The registered application must have reading permission to Azure Monitor (e.g., Subscriptions -> your_subscription -> Access control (IAM) -> Role assignments -> Add -> Add role assignment -> Role : "Monitoring Reader", Select:  my_app)
 
+* If using managed identities:
+  * The VM running the azure-metrics-exporter must have reading permission to Azure Monitor (e.g., Subscriptions -> your_subscription -> Access control (IAM) -> Role assignments -> Add -> Add role assignment -> Role : "Monitoring Reader", Select:  my_vm)
```
[the remark this answers is not in the document]
Comment (shoudusse) [reply README.md]: Hello Brian
I don't really understand what you mean by "your vs my"

File: README.md (hunk 1 above)
Comment (shoudusse) [reply README.md]: Ok, I setted a better explanation at the top.

File: azure.go (hunk 3 above)
Comment (shoudusse) [reply azure.go]: No really an auth but a token retrieval.
Fixed the error message.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (brian-brazil) [reply README.md]: your_subscription vs my_vm

File: README.md [hunk 5]
```diff
@@ -37,11 +37,19 @@ This exporter requires a configuration file. By default, it will look for the az
...
+
 ### Example azure-metrics-exporter config
 
+`authentication_method` can be either `OAuth` or `ManagedIdentity`. Default is `Oauth`
```
Comment (brian-brazil) [inline README.md]: This should only be about the example config, not general reference docs.

File: azure.go [hunk 6]
```diff
@@ -183,14 +183,27 @@ func NewAzureClient() *AzureClient {
...
-	form := url.Values{
-		"grant_type":    {"client_credentials"},
-		"resource":      {sc.C.ResourceManagerURL},
-		"client_id":     {sc.C.Credentials.ClientID},
-		"client_secret": {sc.C.Credentials.ClientSecret},
-	}
-	resp, err := ac.client.PostForm(target, form)
+	var resp *http.Response
+	var err error
+	if sc.C.AuthenticationMethod == "ManagedIdentity" {
```
Comment (brian-brazil) [inline azure.go]: Can we auto-detect this based on what is passed?

Comment (shoudusse) [reply azure.go]: Yes, good idea.
Did it checking if `client_id` is set.

File: README.md (hunk 5 above)
Comment (shoudusse) [reply README.md]: I removed it from the example as this configuration is no more needed.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (shoudusse) [reply README.md]: Ok, I fixed the inconsistency for both parts

File: README.md [hunk 7]
```diff
@@ -59,6 +64,7 @@ You can find endpoints for national clouds [here](http://www.azurespeed.com/Info
 ```
 active_directory_authority_url: "https://login.microsoftonline.com/"
 resource_manager_url: "https://management.azure.com/"
+authentication_method: "Oauth"
```
Comment (brian-brazil) [inline README.md]: This isn't needed.

Comment (shoudusse) [reply README.md]: Argh, I forgot it.
Ok, removed.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**